### ISSUE TITLE
Getting "Warning: Some code is trying to do a URL redirect" when using WP CLI

### ISSUE THREAD

**Issue by ashucg** [W3EDGE/w3-total-cache#305]
As the title suggests, I am getting “Warning: Some code is trying to do a URL redirect” while trying to run cron jobs using WP CLI.

**CLI Command:** `/usr/local/bin/wp --path=/path/to/wordpress-site/ cron event run --due-now`

**Plugin Version:** 0.15.2

---

### Backtrace
```
#0  WP_CLI\Utils\wp_redirect_handler(https://example.com) called at [/path/to/wordpress-site/wp-includes/class-wp-hook.php:289]
#1  WP_Hook->apply_filters(https://example.com, Array ([0] => https://example.com,[1] => 301)) called at [/path/to/wordpress-site/wp-includes/plugin.php:206]
#2  apply_filters(wp_redirect, https://example.com, 301) called at [/path/to/wordpress-site/wp-includes/pluggable.php:1257]
#3  wp_redirect(https://example.com, 301) called at [/path/to/wordpress-site/wp-content/plugins/w3-total-cache/PgCache_Plugin.php:218]
#4  W3TC\PgCache_Plugin->redirect_on_foreign_domain() called at [/path/to/wordpress-site/wp-includes/class-wp-hook.php:287]
#5  WP_Hook->apply_filters(, Array ([0] => )) called at [/path/to/wordpress-site/wp-includes/class-wp-hook.php:311]
#6  WP_Hook->do_action(Array ([0] => )) called at [/path/to/wordpress-site/wp-includes/plugin.php:478]
#7  do_action(init) called at [/path/to/wordpress-site/wp-settings.php:546]
#8  require(/path/to/wordpress-site/wp-settings.php) called at [phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php:1237]
#9  WP_CLI\Runner->load_wordpress() called at [phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php:1158]
#10 WP_CLI\Runner->start() called at [phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Bootstrap/LaunchRunner.php:23]
#11 WP_CLI\Bootstrap\LaunchRunner->process(WP_CLI\Bootstrap\BootstrapState Object ([WP_CLI\Bootstrap\BootstrapStatestate] => Array ())) called at [phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/bootstrap.php:74]
#12 WP_CLI\bootstrap() called at [phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/wp-cli.php:27]
#13 include(phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/wp-cli.php) called at [phar:///usr/local/bin/wp/php/boot-phar.php:11]
#14 include(phar:///usr/local/bin/wp/php/boot-phar.php) called at [/usr/local/bin/wp:4]
```

*I have replaced website URL with `example.com`*

**Comment by mavas84** [comment]
Hello @ashucg 

Thank you for taking the time to report this. 
Seems like a valid issue and the devs will try to replicate this behavior and get back to you once the tests are completed.
Thanks!

**Comment by mo2menelzeiny** [comment]
I'm facing the same issue as well

**Comment by mavas84** [comment]
Hello @ashucg  

Thank you once again for taking the time to report this.
Big shout-out to @mo2menelzeiny for creating the pull request.
This is now fixed in the latest 2.1.6 release!
